modules_core/com.etendorx.auth/src/main/java/com/etendorx/auth/auth/AuthService.java

```diff
@@ -85,11 +109,75 @@ public UserModel validateCredentials(String username, String password) {
     return userModel;
   }
 
-  public Claims generateUserClaims(UserModel userModel) {
+  public String validateService(UserModel userModel, JwtRequest authRequest) {
+    try {
+      if (isSuperUser(userModel)) {
+        return null;
+      }
+
+      List<HashMap<String, String>> accessList = userModel.getETRXRxServicesAccessList();
+      if (accessList.isEmpty()) {
+        throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                "ServicesAccessList can not be empty. This user do not have access to this service");
+      }
+
+      String rxServiceId = accessList.get(0).get("rxServiceId");
+      HttpHeaders headers = new HttpHeaders();
+      headers.add(HEADER_TOKEN, token);
+
+      ResponseEntity<String> rxServiceResponse = serviceClient.searchServiceByServiceId(rxServiceId, true, headers);
+      HttpStatus statusCode = rxServiceResponse.getStatusCode();
+      if (statusCode != HttpStatus.OK) {
+        throw new ResponseStatusException(statusCode, "Unsuccessful DAS connection.");
+      }
+
+      String rxServiceJson = rxServiceResponse.getBody();
+      ObjectMapper mapper = new ObjectMapper();
+      ServicesAccessModel servicesAccessModel = mapper.readValue(rxServiceJson, ServicesAccessModel.class);
+      if (servicesAccessModel == null) {
+        throw new ResponseStatusException(HttpStatus.NO_CONTENT, "DAS server respond with not content.");
+      }
+
+      List<RxServices> rxServicesList = servicesAccessModel.get_embedded().getRxServiceses();
+      if (rxServicesList.isEmpty()) {
+        throw new ResponseStatusException(HttpStatus.NO_CONTENT, "RxServices list is empty.");
+      }
+
+      RxServices rxService = rxServicesList.get(0);
+      if (!StringUtils.equals(rxService.getSearchkey(), authRequest.getService())) {
+        throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, UNAUTHORIZED_SERVICE_MESSAGE);
+      }
+      if (! StringUtils.equals(rxService.getSecret(), authRequest.getSecret())) {
+        throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, UNAUTHORIZED_SECRET_MESSAGE);
+      }
+      return rxService.getSearchkey();
+    } catch (RestClientException e) {
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Error during DAS connection.", e);
+    } catch (JsonProcessingException e) {
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Error during JSON parsing.", e);
+    }
+  }
+
+
+  /**
+   * Generates JWT claims for a user.
+   *
+   * @param userModel the user model
+   * @return the JWT claims
+   */
+  public Claims generateUserClaims(UserModel userModel, String searchKey) {
     Claims claims = new DefaultClaims();
+    HashMap<String, String> servicesAccess = new HashMap<>();
+    if (!isSuperUser(userModel)) {
+      servicesAccess = userModel.getETRXRxServicesAccessList().get(0);
```

Comment:
if userModel.getETRXRxServicesAccessList() is empty, this sentence return indexOutOfBound

Reply:
indexOutOfBound is validated in previous method (validateService)